### PR TITLE
rosettacode: Added semordnilap task

### PR DESCRIPTION
**Pull request by R4PaSs**
See: http://rosettacode.org/wiki/Semordnilap

### CODE REVIEW

File: examples/rosettacode/semordnilap.nit [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Simple Rosettacode task for the detection of semordnilaps in a text file
```
Comment (privat) [inline examples/rosettacode/semordnilap.nit]: link the webpage, easier to understand what is the task (and what is a semordnilaps)

File: tests/sav/semordnilap.res [hunk 2]
```diff
@@ -0,0 +1 @@
+Runtime error: Assert 'index' failed (../lib/standard/collection/array.nit:320)
```
Comment (privat) [inline tests/sav/semordnilap.res]: wat?

File: examples/rosettacode/semordnilap.nit [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+for i in words do
+	var revi = i.reversed
+	if i != revi and words.has(revi) and not semordnilaps.has(i) then
+		semordnilaps.add i
+		semordnilaps.add revi
+		pairs[i] = revi
+	end
+end
+
+print "Found {pairs.length} semordnilaps in {args[0]}"
```
Comment (Morriar) [inline examples/rosettacode/semordnilap.nit]: [Imgur](http://i.imgur.com/q9L0FSk.png)

**Comment by R4PaSs** [comment]
Reroll: Added link to semordnilap and fixed the crash on empty args

File: examples/rosettacode/semordnilap.nit [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+if args.length < 1 then
+	print "Usage: ./semordnilap file"
+	exit(1)
+end
+
+var fr = new FileReader.open(args[0])
+var words = new HashSet[String]
+
+for i in fr.each_line do words.add i.trim
```
Comment (privat) [inline examples/rosettacode/semordnilap.nit]: What about using Path?

~~~nit
for l in args[0].to_path.each_line do words.add l.trim
~~~

File: examples/rosettacode/semordnilap.nit [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Simple Rosettacode task for the detection of semordnilaps in a text file
+#
+# SEE: http://rosettacode.org/wiki/Semordnilap
+module semordnilap
+
+if args.length < 1 then
```
Comment (privat) [inline examples/rosettacode/semordnilap.nit]:
~~~nit
if args.is_empty then
~~~

File: examples/rosettacode/semordnilap.nit (hunk 4 above)
Comment (privat) [inline examples/rosettacode/semordnilap.nit]: also name a variable to avoid using a semantic-less `args[0]`.

~~~nit
var filename = args[0]
~~~

**Comment by R4PaSs** [comment]
Reroll: Fixed @privat's remarks

**Pull request merged**